src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -7,19 +7,22 @@
 using Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
 using Microsoft.AspNetCore.App.Analyzers.Infrastructure;
 using Microsoft.CodeAnalysis;
+using Microsoft.AspNetCore.Http;
 using WellKnownType = Microsoft.AspNetCore.App.Analyzers.Infrastructure.WellKnownTypeData.WellKnownType;
 
 namespace Microsoft.AspNetCore.Http.Generators.StaticRouteHandlerModel;
 
 internal class EndpointParameter
 {
-    public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTypes)
+    public EndpointParameter(Endpoint endpoint, IParameterSymbol parameter, WellKnownTypes wellKnownTypes)
```

Comment:
Why not just pass the HTTP method here?

Reply:
Initially I was thinking I might need some per endpoint state captured in a variable (around disabling parameter inferencing for POSTs). Ended up not needing it, but may still need it when we get to MapMethods etc.